### PR TITLE
Rework to for ansible modules

### PR DESCRIPTION
**Pull request by otcbot**
Some changes to introduce proper modules and correspondingly to make it properly reusable

### CODE REVIEW

**Comment by gtema** [comment]
@anton-sidelnikov , @tibeer  I am still on it and will add also "people" management module after the weekend.
With modules approach we have a better performance and improved usability while keeping structure of the configuration.

I am explicitly not trying to have a module per action type, since in this case depending on the size of the config we have too much of overhead of Ansible for individual module invocation. Instead single module `repositories`, `teams`, `people` read the whole config and process it at once.

**Comment by tibeer** [comment]
@gtema sounds good for me. Might also help prevent flooding the GitHub API and getting blocked.

File: plugins/modules/teams.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/python
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+module: repository
```
Comment (anton-sidelnikov) [inline plugins/modules/teams.py:8]: teams

File: plugins/modules/repositories.py [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+    def _is_repo_access_update_needed(self, current, target):
+        if current != target:
+            return True
+        for priv in ['maintain', 'pull', 'push', 'admin']:
+            if current[priv] != target[priv]:
+                return True
+        return False
+
+    def _get_privs(self, mapping):
+        """Convert teams/collabirators mapping into entity/priv mapping"""
```
Comment (anton-sidelnikov) [inline plugins/modules/repositories.py:144]: collaborators

**Review by anton-sidelnikov — APPROVED** [review]

**Pull request merged**